frontend/src/store/course/actions.ts

```diff
@@ -272,17 +272,24 @@ const updateIsMentorChoosingEnabled = createAsyncThunk<
   async (id, { extra, getState }) => {
     const {
       auth: { user },
-      course: { mentors },
+      course: { mentors, course },
     } = getState();
 
+    if (!course) {
+      return false;
+    }
+
+    const { coursesApi } = extra;
+
     const isMentor = mentors.some(
       (mentor) => mentor.id === (user as UserWithPermissions).id,
     );
-    const hasMentor = await extra.coursesApi.checkHasMentor({
+    const hasMentor = await coursesApi.checkHasMentor({
       courseId: id,
     });
+    const hasCategory = Boolean(course.category);
 
-    const canChooseMentor = !(isMentor || hasMentor);
+    const canChooseMentor = !(isMentor || hasMentor || !hasCategory);
```

Comment:
I think this syntax is more clear
```suggestion
    const canChooseMentor = isMentor || hasMentor || !hasCategory;
```

Reply:
@Liza-Veis 
But your check would return different result 🤔 
I think same check without `!(<conditions>)` would look like this
```
    const canChooseMentor = !isMentor && !hasMentor && hasCategory;
```

Reply:
yeah, I've already changed it 😅